src/sentry/models/notificationmessage.py

```diff
@@ -0,0 +1,51 @@
+from django.db.models import DateTimeField
+from django.utils import timezone
+
+from sentry.backup.scopes import RelocationScope
+from sentry.db.models import (
+    CharField,
+    FlexibleForeignKey,
+    JSONField,
+    Model,
+    region_silo_only_model,
+    sane_repr,
+)
+
+
+@region_silo_only_model
+class NotificationMessage(Model):
+    """
+    Data model represents the aggregate for an entire notification message.
+    A notification message must always have the request and response that it took, along with the resulting identifier
+    that was sent back.
+
+    The data model hold the singular gateway for both Metric, and Issue, Alerts.
+    Following the specific data relationship, you should be able to find the specific rule, organization, project, and
+    incident or group event that this notification message was enacted.
+
+    If any notification message is in relation to another notification message, you should be able to find the original
+    notification message through the parent notification message relation.
+    """
+
+    __relocation_scope__ = RelocationScope.Excluded
+
+    request = JSONField(null=False)
+    response = JSONField(null=False)
```

Comment:
Storing the request and response is quite a lot of data. Is this just for debugging? Would logging be sufficient if that's the case?

Reply:
Yeah super fair question! We've had requests from users and our ops team asking which alerts are failing, and to help give that visibility. The first step was to just create a dashboard and give the ops team that insight, then eventually move that into the product to show "These X alerts are failing, take a look" or something. More context at: https://sentry.slack.com/archives/C04KZQBNQ2U/p1707170008881059

What I could do instead is just store the response/error code, along with the response details, only when the request fails. This way we reduce it down to the failed notifications which is our primary concern anyways. Does that sound fair?

Reply:
Yeah I'd agree with that, just log on error should reduce the volume a lot, that seems reasonable.

Reply:
Updated!